core/cap-0033.md

```diff
@@ -356,38 +357,60 @@ struct InnerTransactionResult
+`balance < (2 + numSubEntries + numSponsoring - numSponsored) * baseReserve + liabilities.selling`.
 
-#### Sponsorship Bookkeeping
-When account `A` is-sponsoring-future-reserves-for account `B`, any reserve
+#### Sponsoring-Future-Reserves
+When account `A` is sponsoring-future-reserves for account `B`, any reserve
 requirements that would normally accumulate on `B` will instead accumulate on
-`A` as reflected in `numSponsoring`. Both accounts are guaranteed to exist, but
-this can fail. If such a change would cause `A` to pass below the reserve
-requirement, then the operation fails. The fact that these reserves are being
-provided by another account will be reflected on `B` in `numSponsored`.
-
-When a sponsored ledger entry or sub-entry is removed, `numSponsoring` is
-decreased on the sponsoring account and `numSponsored` is decreased on the
-sponsored account. Both accounts are guaranteed to exist, so this cannot fail.
+`A`. This is achieved by updating `A.numSponsoring` and `B.numSponsored` (unless
+the reserve requirement is accumulating due to a `ClaimableBalance` entry).
+
+An account `A` begins sponsoring-future-reserves for an account `B` upon a
+successful `BeginSponsoringFutureReservesOp` with `sourceAccount = A` and
+`sponsoredID = B`. `A` stops sponsoring-future-reserves for `B` upon a
+successful `EndSponsoringFutureReserves` with `sourceAccount = B`. These are the
+only two operations which can impact the sponsoring-future-reserves state of an
+account.
+
+#### Sponsoring-Future-Reserves Invariants
+There are two invariants related to sponsoring-future-reserves:
+- If an account `A` is sponsoring-future-reserves for an account `B`, then `B`
+  is not sponsoring-future-reserves for any account. Specifically, this
+  prevents an account from sponsoring-future-reserves for itself.
+- If an account `A` is sponsoring-future-reserves for an account `B`, then `A`
+  exists. The converse need not be true (this facilitates creating a sponsored
+  account).
+
+#### Sponsorship Invariants
+There are three invariants related to sponsorships:
+- (Global) The sum of all `numSponsoring` is equal to the sum of all
+  `numSponsored` plus the sum of all `ClaimableBalance.claimants.size()`.
+- (Local) For an account `A`, `numSponsoring` is equal to the count of all
+  `LedgerEntry le` with `le.ext.v1().sponsoringID == A` weighted by the number
+  of reserves required for that `LedgerEntry` (2 for `ACCOUNT`, `claimants.size()`
+  for `CLAIMABLE_BALANCE`, 1 otherwise) plus the count of all `A` in
+  `signerSponsoringIDs`.
+- (Local) For an account `A`, `numSponsored` is equal to the count of all
+  `LedgerEntry le` which are sub-entries of `A` (including `A` itself) that have
+  `le.ext.v1().sponsoringID != null` weighted by the number
+  of reserves required for that `LedgerEntry` (2 for `ACCOUNT`, `claimants.size()`
+  for `CLAIMABLE_BALANCE`, 1 otherwise) plus the count of all `s` in
+  `A.ext.v1().ext.v2().signerSponsoringIDs` with `s != null`.
```

Comment:
```suggestion
  `le.ext.v1().sponsoringID != null` weighted by the number of reserves 
  required for that `LedgerEntry` (2 for `ACCOUNT`, 1 otherwise) plus the count of all `s` 
  in `A.ext.v1().ext.v2().signerSponsoringIDs` with `s != null`. Note that
  `Claimable balances` are not sub-entries, so they do not affect `numSponsored`.
```